request-response/src/recipient_source.rs

```diff
@@ -7,7 +7,7 @@ use super::request::Request;
 /// expect responses from. In `HotShot` this would go on top of the [`Membership`] trait and determine
 /// which nodes are able (quorum/DA) to respond to which requests
 #[async_trait]
-pub trait RecipientSource<K: SignatureKey + 'static>: Send + Sync + 'static {
+pub trait RecipientSource<R: Request, K: SignatureKey + 'static>: Send + Sync + 'static {
     /// Get all the recipients that the specific request should expect responses from
-    async fn get_recipients_for<R: Request>(&self, request: &R) -> Vec<K>;
+    async fn get_recipients_for(&self, request: &R) -> Vec<K>;
```

Comment:
Based on the doc comment how about `expected_responders`? I think this labels the role more explicitly because there are recipients of requests, but also recipients of responses. I know the `Request` type are there so it's still deducible but anyway.